src/modules/SdnDiag.Server.psm1

```diff
@@ -3398,3 +3400,141 @@ function Get-SdnVMSwitch {
+    }
+    else {
+        $ncRestParams.Add('NcRestCredential', $NcRestCredential)
+    }
+
+    $repairPortProfileParams = @{
+        VMName      = $VMName
+        MacAddress  = [System.String]::Empty
+        ProfileId   = [System.Guid]::Empty
+        ProfileData = 1
+    }
+
+    try {
+        # format the mac address to match the format used by Network Controller
+        # then invoke request to retrieve the network interfaces that match the MAC address
+        # we need the InstanceId of the network interface
+        $formattedMacAddress = Format-SdnMacAddress -MacAddress $MacAddress
+        $networkInterfaces = Get-SdnResource @ncRestParams -ErrorAction Stop | Where-Object { $_.properties.privateMacAddress -eq $formattedMacAddress }
+        if ($null -eq $networkInterfaces) {
+            throw New-Object System.ArgumentException("Unable to locate NetworkInterface with MAC Address $formattedMacAddress.")
+        }
+
+        $repairPortProfileParams.MacAddress = $formattedMacAddress
+        $repairPortProfileParams.ProfileId = $networkInterfaces.InstanceId
+
+        # check to see if the Hyper-V host is local or remote host
+        if (Test-ComputerNameIsLocal -ComputerName $HyperVHost) {
+            $vmNetworkAdapters = Get-SdnVMNetworkAdapterPortProfile -VMName $VMName -ErrorAction Stop
+            $currentPortProfileSettings = $vmNetworkAdapters | Where-Object {$_.MacAddress -eq $formattedMacAddress}
+        }
+        else {
+            $repairPortProfileParams.Add('HyperVHost', $HyperVHost)
+            $repairPortProfileParams.Add('Credential', $Credential)
+
+            $currentPortProfileSettings = Invoke-SdnCommand -ComputerName $HyperVHost -Credential $Credential -ScriptBlock {
+                param($vmName, $macAddress)
+
+                $vmNetworkAdapters = Get-SdnVMNetworkAdapterPortProfile -VMName $vmName
+                return ($vmNetworkAdapters | Where-Object {$_.MacAddress -eq $macAddress})
+            } -ArgumentList @($VMName, $formattedMacAddress) -ErrorAction Stop
+        }
+        if ($null -ieq $currentPortProfileSettings) {
+            throw New-Object System.ArgumentException("Unable to locate Port Profile for VM $VMName with MAC Address $formattedMacAddress.")
+        }
+
+        # ensure that the profile id matches the instance id of the networkinterface
+        $formattedCurrentProfileId = [System.Guid]::Parse($currentPortProfileSettings.ProfileId).Guid
+        if ($formattedCurrentProfileId -ne $repairPortProfileParams.ProfileId) {
+            $repairPortProfileParams.ProfileId = $networkInterfaces.InstanceId
+            "Current ProfileId [{0}] does not match expected ProfileId [{1}]." -f $formattedCurrentProfileId, $repairPortProfileParams.ProfileId | Trace-Output -Level:Information
+            $repairRequired = $true
+        }
+
+        # ensure that the profile data is set to 1 (VfpEnabled)
+        if ($currentPortProfileSettings.ProfileData -ne 1) {
```

Comment:
Would profile data ever not be 1? On my env, external MUX NICs have profile data 7, HNVPA MUX NICs have profile data 2 (Looking through the SDN code, it looks like 2 was meant for MAS to use for tenant VMs. Not 100% sure what the difference is between 1 and 2).

Reply:
I added a block if networkInterface is associated to Gateway or LoadBalancerMux (Internal/External) interface. In addition, AzLocal supports LogicalNetworks w/ DHCP Enabled workloads, which can be identified by the 'UnManaged' value of the privateIPAllocationMethod key. In this scenario, VFP ProfileData would be 6 to support this config.